### PR TITLE
Slice and grid

### PR DESCRIPTION
**Pull request by Lokathor**
* New: VolGrid2d
* New VolGrid3d
* New: VolRegion

### CODE REVIEW

**Comment by Lokathor** [comment]
@nicopap i can't assign you as a reviewer since i guess you're not in the org but you did the initial 2d work so please have a look and point out anything that looks out of place.

(the PR was all done with github's web interface rather than in an IDE so there could be things that the CI isn't seeing.)

**Review by nicopap — APPROVED** [review]

File: src/volgrid3d.rs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  #[inline]
+  #[must_use]
+  pub const fn get_frame(self, z: usize) -> Option<VolGrid2d<T, R, W, WIDTH, HEIGHT>> {
+    if z < FRAMES {
+      // SAFETY:
+      // - `y < HEIGHT`
+      // - `VolGrid2d::new` safety condition guarantees that all addresses
+      //   constructible for `VolBlock<T, WIDTH>` are valid `VolAddress`,
+      //   which is the safety condition of `VolBlock::new`.
+      Some(unsafe { VolGrid2d { base: self.base.add(z * (WIDTH * HEIGHT)) } })
```
Comment (Imberflur) [inline src/volgrid3d.rs:48]: This comment hasn't been updated to match that this is for `VolGrid3d`

Comment (Lokathor) [reply src/volgrid3d.rs:48]: Good catch, updated in https://github.com/rust-console/voladdress/commit/94c668498311d451c88ad5245700bc184a5d81df

**Comment by Lokathor** [comment]
When I started to think more about where I'd need to use this in the `gba` crate, I adjusted the "3d" version to always take a stride.

https://github.com/rust-console/voladdress/commit/2a25ad14c850fd7553df244af73c890230b955a0

I think it will appear infrequently enough that having to manually provide a stride is not a massive cost.

**Pull request merged**